src/components/login/registerForm.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect } from 'react';
+import { TextField, Button, CircularProgress, Snackbar, Typography } from '@material-ui/core';
+import authStyles from './auth-style';
+import { Redirect } from 'react-router-dom';
+import Alert from '@material-ui/lab/Alert';
+
+const RegisterForm = function() {
+  const classes = authStyles();
+
+  const [username, setUsername] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const [showRegisterFail, setShowRegisterFail] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  // TODO: remove this after implementing the authentication service
+  const [redirect, setRedirect] = React.useState(false);
+
+  useEffect(() => {
+    if (
+      username.trim() &&
+      password.trim() &&
+      firstName.trim() &&
+      lastName.trim() &&
+      confirmPassword.trim()
+    ) {
+      setIsButtonDisabled(false);
+    } else {
+      setIsButtonDisabled(true);
+    }
+  }, [username, password, firstName, lastName, confirmPassword]);
+
+  function isFormValid(): boolean {
+    let isValid = true;
+    if (password !== confirmPassword) {
+      setShowRegisterFail(true);
+      setErrorMessage('Passwords do not match');
+      isValid = false;
+    }
+    return isValid;
+  }
+
+  function handleRegisterFail(err: string): void {
+    setIsSubmitting(false);
+    setErrorMessage(err);
+    setShowRegisterFail(true);
+  }
+
+  function handleRegisterSuccess(): void {
+    // TODO replace this with the register logic
+    setTimeout(() => {
+      setIsSubmitting(false);
+      setRedirect(true);
+    }, 1000);
+  }
+
+  function submit(event: React.FormEvent<HTMLFormElement>): void {
+    event.preventDefault();
+
+    if (isFormValid()) {
+      // props.auth.register(username, firstName, lastName, password, confirmPassword);
+      // TODO replace this logic with the a call to the authentication service
+      if (username === 'user') {
+        setIsSubmitting(true);
+        handleRegisterSuccess();
+      } else {
+        // Replace with backend error message
+        handleRegisterFail('Has been an error creating your account');
+      }
+    }
+  }
+
+  return (
+    <div className={`${classes.flexColumnContainer} ${classes.fullPage}`}>
+      {!!redirect && <Redirect to="/login" />}
+      <div className={`${classes.flexColumnContainer} ${classes.authContainer}`}>
+        <h1 className={classes.authTitle}>RoMi Dashboard</h1>
```

Comment:
Use `Typography` to keep things consistent.

Reply:
Done, thanks!